api/src/main/java/org/pmiops/workbench/workspaceadmin/WorkspaceAdminController.java

```diff
@@ -34,18 +47,43 @@
 
   @Autowired
   public WorkspaceAdminController(
+      CloudMonitoringService cloudMonitoringService,
       FireCloudService fireCloudService,
       LeonardoNotebooksClient leonardoNotebooksClient,
       WorkspaceAdminService workspaceAdminService,
       WorkspaceMapper workspaceMapper,
       WorkspaceService workspaceService) {
+    this.cloudMonitoringService = cloudMonitoringService;
     this.fireCloudService = fireCloudService;
     this.leonardoNotebooksClient = leonardoNotebooksClient;
     this.workspaceAdminService = workspaceAdminService;
     this.workspaceMapper = workspaceMapper;
     this.workspaceService = workspaceService;
   }
 
+  /** */
+  @Override
+  @AuthorityRequired({Authority.WORKSPACES_VIEW})
+  public ResponseEntity<CloudStorageTraffic> getCloudStorageTraffic(String workspaceNamespace) {
+    CloudStorageTraffic response = new CloudStorageTraffic().receivedBytes(new ArrayList<>());
+
+    for (TimeSeries timeSeries :
```

Comment:
At least one of these could probably done as a stream instead of an old-school foreach loop. Though I do vaguely recall nested streams being obnoxious to deal with.